### ISSUE TITLE
[Bug]: UID Configuration in Jsonnet File Not Working and Causes Error

### ISSUE THREAD

**Issue by BlaCkinkGJ** [grafana/grizzly#460]
### Grizzly Version

v0.4.0

### Expected Behavior

- If the UID is not specified, the system should automatically add the UID to the Jsonnet file, similar to how it was handled in version 0.3.1.
- If the UID is specified, it should be explicitly set.

### Actual Behavior

- Regardless of whether the UID was specified in the Jsonnet file, it did not function correctly and displayed the following error message:
- Even when the UID was specified, the error message indicated that the UID wasn't configured.

> I generated the following error message using the source code provided in the "Steps to Reproduce" section.

<img width="1389" alt="image" src="https://github.com/grafana/grizzly/assets/16631264/2e54b661-2212-41bf-bd45-354e0d58950d">

### Steps to Reproduce

1. **Prepare the Jsonnet File**:
   - Create a Jsonnet file to generate a dashboard using Grizzly. We will use [grafonnet-7.0](https://github.com/grafana/grafonnet-lib/blob/master/grafonnet-7.0/DOCS.md) to configure the Jsonnet file. Name the test file `test.jsonnet`.
   ```jsonnet
   local grafana = import 'grafonnet-7.0/grafana.libsonnet';
   local dashboard = grafana.dashboard;

   {
     grafanaDashboardFolder:: 'Test Dashboard Folder',
     grafanaDashboards:: {
       ['test_dashboard']: dashboard.new(title="hello", uid="hello"),
     },
   }
   ```

2. **Set Configuration Values**:
   - Configure the `GRAFANA_URL` and `GRAFANA_TOKEN` values.

3. **Apply the Jsonnet File**:
   - Run the command `grr apply test.jsonnet`.

**Note**: In my local environment, I was able to resolve this issue by either:
- Removing the following code segment:
  [Link to the specific line in grizzly.jsonnet](https://github.com/grafana/grizzly/blob/44fdd721aa4e9b0abc1b132e4dd907e388dd473d/pkg/grizzly/grizzly.jsonnet#L57)
- Adding logic to attach the UID part from version 0.3.1:
  [Link to the specific lines in dashboard-handler.go](https://github.com/grafana/grizzly/blob/115008d77e1e575140a46a7b6ba79e5d4b0210c3/pkg/grafana/dashboard-handler.go#L71-L74)

**Comment by malcolmholmes** [comment]
This really isn't straightforward, and there isn't a right answer. Say someone wants a dashboard delivering to Grafana *without* a UID - if Grizzly added the `metadata/name` as UID, it would then be shipping a different dashboard to the one the user expected, resulting in unexpected behaviour.

Using validation to confirm that `metadata/name` and UID both match, and erroring out otherwise is the simplest and cleanest of a bad set of options.

**Comment by BlaCkinkGJ** [comment]
Thank you for looking into this issue.

I would like to point out that setting the `UID` value does not seem to work as expected, regardless of the validation routine.

For instance, even when I configure `metadata/name` to "hello" and set the `UID` to "hello," I still receive the same error message. It appears that the `UID` is always empty, causing the validation routine to consistently fail.

In summary, the `UID` value setting does not appear to be functioning correctly.

```
Dashboard.hello failed: uid "" and name "hello", don't match.
```

**Comment by malcolmholmes** [comment]
can you give me a simple snippet that demonstrates this case?

**Comment by BlaCkinkGJ** [comment]
You can reproduce this issue by using the following dashboard.jsonnet file.

```jsonnet
local grafana = import 'grafonnet/grafana.libsonnet';

{
  grafanaDashboards:: {
    empty_dashboard: grafana.dashboard.new(title="hello", uid="world"),
  },
}
```

Its result:
<img width="591" alt="image" src="https://github.com/user-attachments/assets/2807ee8b-a013-4687-8c44-2a7b1509f604">

The commands are as follows:

```bash
mkdir simple-dashboard
cd simple-dashboard
jb init
jb install https://github.com/grafana/grafonnet-lib/grafonnet
vi dashboard.jsonnet # put the content above
grr apply dashbaord.jsonnet
```

Just in case, I updated my grizzly version from v0.4.0 to v0.4.3, but I still got the same error message.

```bash
2024/07/17 10:27:36 grr version v0.4.3
```

**Comment by minhthong582000** [comment]
I'm encountering the same error using the example in the `examples` folder:

```bash
# grr version 0.4.3
cd examples/
grr apply grr.jsonnet
```

It seems like the error is coming from these `uid` validation logic when getting dashboard from Grafana which I think is redundant, as `resource.name()` already serves as a valid uid.

https://github.com/grafana/grizzly/blob/main/pkg/grafana/dashboard-handler.go#L91-L94

I've created a PR to remove the check: https://github.com/grafana/grizzly/pull/466. Please have a look and let me know if it have any problems.

**Comment by minhthong582000** [comment]
I am also facing the same issue.

As pointed out in the description, the `grizzly.jsonnet` file sets the dashboard `uid` to an empty string (https://github.com/grafana/grizzly/blob/main/pkg/grizzly/grizzly.jsonnet#L41-L42) when parsing the jsonnet input file, making the validation logic redundant. The `resource.Name()` function then serves as a valid `uid`.

If the intention is to always keep the `uid` as an empty string, I've created an example PR to remove the validation: https://github.com/grafana/grizzly/pull/466, and it seems to work.

**Comment by malcolmholmes** [comment]
Ahh, thanks, I haven't been looking at `grizzly.jsonnet`. I'm AFK until next week. I can take a look then, unless someone beats me to it.

**Comment by minhthong582000** [comment]
Hi @malcolmholmes, just a friendly reminder, please check this when you get a chance. Thank you! 😄
